src/types/quota.rs

```diff
@@ -0,0 +1,246 @@
+use crate::error::{Error, ParseError};
+use crate::parse::{parse_many_into2, parse_until_done, MapOrNot, MapOrNot2};
+use crate::types::UnsolicitedResponse;
+use imap_proto::Response;
+use ouroboros::self_referencing;
+use std::borrow::Cow;
+use std::fmt::{Debug, Display, Formatter};
+use std::sync::mpsc;
+
+/// From [SETQUOTA Resource limit](https://datatracker.ietf.org/doc/html/rfc2087#section-4.1)
+///
+/// Used by [`Session::set_quota`]
+#[derive(Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub struct QuotaResourceLimit<'a> {
+    /// The resource type
+    pub name: QuotaResourceName<'a>,
+    /// The amount for that resource
+    pub amount: u64,
+}
+
+impl<'a> QuotaResourceLimit<'a> {
+    /// Creates a new [`QuotaResourceLimit`]
+    pub fn new(name: QuotaResourceName<'a>, amount: u64) -> Self {
```

Comment:
How about taking `impl Into<QuotaResourceName<'a>>` here, and then implementing `From<str>` for `QuotaResourceName`?